metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -976,6 +1005,49 @@ object MetalsEnrichments
       trailingTokens.find(predicate)
   }
 
+  implicit class XtensionTreeBraceHandler(stat: Tree) {
+
+    /**
+     * Check if it's possible to use braceless syntax and whether
+     * it's the preferred style in the file.
+     */
+    def canUseBracelessSyntax(source: String): Boolean = {
+
+      def allowBracelessSyntax(tree: Tree) = tree.origin match {
+        case p: Parsed => p.dialect.allowSignificantIndentation
+        case _ => false
+      }
+
+      def isNotInBraces(t: Tree): Boolean = {
+        t match {
+          case _: Template | _: Term.Block => source(t.pos.start - 1) != '{'
```

Comment:
This should be:
```suggestion
          case _: Template | _: Term.Block if  t.pos.start < source.length => source(t.pos.start) != '{'
```
that's why the tests changed.